### DISCUSSION TITLE
categories.json cannot work correctly

### DISCUSSION THREAD

**Discussion by gitwan** [allure-framework/allure2#2294] [Questions & Support]
### Describe the Bug

1. based on this doc: https://allurereport.org/docs/how-it-works-categories-file/, and customize categories.json as below:
```json
[
  {
    "name": "Product defects - User",
    "matchedStatuses": [
      "failed"
    ],
    "messageRegex": ".*(payment|account).*"
  }
]
```
3. run the ./allure generate and ./allure serve command, found the result didn't match the condition:  `"messageRegex": ".*(payment|account).*"`

### Steps to Reproduce

1. based on this doc: https://allurereport.org/docs/how-it-works-categories-file/, and customize categories.json as below:
```json
[
  {
    "name": "Product defects - User",
    "matchedStatuses": [
      "failed"
    ],
    "messageRegex": ".*(payment|account).*"
  }
]
```
2. run the ./allure generate and ./allure serve command, found the result didn't match the condition:  `"messageRegex": ".*(payment|account).*"`

### Expected Behaviour

the failed testcase that message includes payment or account will be grouped separately in "Product defects - User",

### What Language are you using?

Java

### What Framework/Allure Integration you are using?

allure

### What version of Allure Integration you are using?

2.21

### What version of Allure Report you are using?

2.21

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by baev** [comment]
Please provide more details on the problem. Screenshots that show the result not matched into a category would help. The repository with a sample project to reproduce the problem would be the best (you can use https://allurereport.org/start/ to generate the template)
